src/analyzing/AST_to_IL.ml

```
@@ -1241,6 +1241,14 @@ and stmt_aux env st =
       let ss, e' = expr_with_pre_stmts env e in
       let lv = lval_of_ent env ent in
       ss @ [ mk_s (Instr (mk_i (Assign (lv, e')) (Related (G.S st)))) ]
+  | G.DefStmt (ent, G.VarDef { G.vinit = None; vtype = Some ty }) -> (
```

I think it's better to write this pattern instead:
```
... vtype = Some { t = G.TyArray ((_, Some e, _), _); _} ...
```
so that if it's not an array type as you want, then we don't end up in your default `[]` case, which is "masking" whatever the default case used to be.